### PR TITLE
Enable 'Equal Earth' projection

### PR DESCRIPTION
**Pull request by apparebit**
[Equal Earth](https://en.wikipedia.org/wiki/Equal_Earth_projection) is an aesthetically pleasing, fairly recent equal-area pseudocylindrical projection, which has has been adopted by the NASA Goddard Institute for Space Studies. I am submitting this pull request after @alexcjohnson [encouraged me to do so](https://community.plotly.com/t/equal-earth-projection/48141/3) in the plotly forum.

### CODE REVIEW

**Comment by archmoj** [comment]
Thanks very much for the PR.

Please add examples of using this new projection in the following mocks: 
1. `test/image/mocks/various_geo_projections.json`
2. `test/image/mocks/geo_fitbounds-locations.json`
3. `test/image/mocks/canada_geo_projections.json`

Then you could collect & commit new baselines from the `ARTIFACTS` tab of `test-baselines` on `CircleCi`.

File: draftlogs/6670_add.md [hunk 1]
```diff
@@ -0,0 +1 @@
+ - Add 'Equal Earth' projection [[#6670](https://github.com/plotly/plotly.js/pull/6670)]
```
Comment (archmoj) [inline draftlogs/6670_add.md:1]:
```suggestion
 - Add 'Equal Earth' projection [[#6670](https://github.com/plotly/plotly.js/pull/6670)], with thanks to @apparebit for the contribution!
```

**Comment by apparebit** [comment]
I think I have the mock JSON files figured out by now. I am *not* a fan of having to manually shift most of the map cells by one position nor of the mocks using Albers USA as a kind of sentinel (surprise!). Having said that, the robotically incrementing numbers mod 8 is long done. I just need CircleCI to provide me with the updated and correct images. After that, the PR should be ready.

Alas, the "webgl-jasmine" tests are incredibly flaky but I cannot do reruns. Even after signing up with CircleCI with my GitHub account, it insists that I need write permissions to the project. Two options:

 1 . You can empower me and I'll persist with reruns until the board is green.
 2. I have to rely on one of you to do this in my stead.

This is obviously your call. Either works for me. Thank you

**Comment by archmoj** [comment]
New baselines are now available at https://output.circle-artifacts.com/output/job/86068e44-8fa5-4ad9-837a-c78716ce6d35/artifacts/0/baselines.tar (or from https://app.circleci.com/pipelines/github/plotly/plotly.js/8921/workflows/5a975e79-dec3-47df-84d2-e0a0d010bcda/jobs/196144/artifacts in case you don't want to unzip them).
Please update those at `test/image/baselines`.
Thank you!

**Comment by apparebit** [comment]
@archmoj, I think I was committing them while you were writing the comment. The versions I pushed to the branch show Equal Earth and properly shift all subsequent cells including the Albers sentinel.

**Comment by archmoj** [comment]
Fantastic. :medal_military: :trophy: :dancer:

**Pull request merged**